tests/generator/generator.cpp

```diff
@@ -2356,4 +2356,93 @@ TEST(Generator, coverage)
+
+    generator->setProfile(nullptr);
+
+    generator->implementationCode();
+
+    // Coverage for various profile settings.
+
+    analyser->addExternalVariable(libcellml::AnalyserExternalVariable::create(model->component("my_component")->variable("eqnEq")));
+
+    analyser->analyseModel(model);
+
+    generator->setModel(analyser->model());
+    generator->setProfile(profile);
+
+    profile->setAcotFunctionString("");
+    profile->setAcothFunctionString("");
+    profile->setAcscFunctionString("");
+    profile->setAcschFunctionString("");
+    profile->setAsecFunctionString("");
+    profile->setAsechFunctionString("");
+    profile->setCommentString("");
+    profile->setCotFunctionString("");
+    profile->setCothFunctionString("");
+    profile->setCscFunctionString("");
+    profile->setCschFunctionString("");
+    profile->setExternalVariableMethodTypeDefinitionString(true, "");
+    profile->setHasConditionalOperator(true);
+    profile->setHasInterface(true);
+    profile->setHasPowerOperator(true);
+    profile->setHasXorOperator(true);
+    profile->setImplementationComputeComputedConstantsMethodString("");
+    profile->setImplementationComputeRatesMethodString(true, "");
+    profile->setImplementationHeaderString("[INTERFACE_FILE_NAME]");
+    profile->setImplementationInitialiseVariablesMethodString(true, true, "");
+    profile->setInterfaceFileNameString("");
+    profile->setInterfaceHeaderString("");
+    profile->setMaxFunctionString("");
+    profile->setMinFunctionString("");
+    profile->setSecFunctionString("");
+    profile->setSechFunctionString("");
+    profile->setVariableInfoEntryString("");
+
+    generator->interfaceCode();
+    generator->implementationCode();
+
+    profile->setArrayElementSeparatorString("");
+    profile->setCommentString("xxx");
+    profile->setOriginCommentString("");
+    profile->setVariableInfoEntryString("xxx");
+
+    generator->implementationCode();
+
+    profile->setArrayElementSeparatorString("xxx");
+    profile->setVariableOfIntegrationVariableTypeString("");
+
+    generator->implementationCode();
+
+    profile->setStateVariableTypeString("");
+    profile->setVariableOfIntegrationVariableTypeString("xxx");
+
+    generator->implementationCode();
+
+    profile->setConstantVariableTypeString("");
+    profile->setStateVariableTypeString("xxx");
+
+    generator->implementationCode();
+
+    profile->setComputedConstantVariableTypeString("");
+    profile->setConstantVariableTypeString("xxx");
+
+    generator->implementationCode();
+
+    profile->setComputedConstantVariableTypeString("xxx");
+    profile->setAlgebraicVariableTypeString("");
+
+    generator->implementationCode();
+
+    profile->setAlgebraicVariableTypeString("xxx");
+    profile->setExternalVariableTypeString("");
+
+    generator->implementationCode();
+
+    profile->setHasXorOperator(false);
+    profile->setXorFunctionString("");
+
+    generator->implementationCode();
```

Comment:
Dump coverage tests for edge cases should be placed in the coverage file.

Reply:
Done in b1d72b8.